packages/xcm-util/README.md

```diff
@@ -9,8 +9,131 @@
 
 ## Installation
 
-TBD
+Before installing `xcm-util` package to your project, you need to setup Yarn 2, due to requirements of the `open-xcm-tools` packages.
+
+### Yarn 2 (Berry) installation
+
+First, set up a global Yarn, which will be used to create local instances:
+
+```bash
+npm install -g yarn
+```
+
+Then you need to locally change **Yarn** version for your project. In project folder:
+
+```bash
+yarn set version berry
+```
+
+After this actions you can use **Yarn** as usually (init project, add dependencies, etc.)
+
+### `xcm-util` installation
+
+```bash
+yarn add @open-xcm-tools/xcm-util
+```
 
 ## Usage
 
-TBD
+For showing all functionality let's take a look to the examples of `Location` and `Asset` entities:
+
+```typescript
+const location: Location = location(1n, [
+  {parachain: 2001n},
+  {palletInstance: 50n},
+  {generalIndex: 1984n},
+]);
+
+const asset: Asset = asset(location(0n, 'here'), fungible(100n));
+```
+
+### Convert XCM version
+
+The examples above are `Location` and `Asset` instance of latest XCM versions.  The package `xcm-util` provides the opportunity to convert current XCM version of instances to specific, for example:
+
+```typescript
+const locationV3: LocationV3 = convertLocationVersion(3, location);
+const assetV3: AssetV3 = convertAssetVersion(3, asset);
+```
+
+Also you can create `VersionedAsset` instance of XCM version 2, for example, and convert it to the latest XCM version, using:
+
+```typescript
+const versionedAssetV2: VersionedAsset = {
+  V2: {
+    concrete: {
+      interior: 'here',
+      parents: 0n,
+    },
+    fun: {
+      fungible: 10n,
+    },
+  },
+};
+
+const asset: Asset = assetIntoCurrentVersion(versionedAssetV2);
+```
+
+### Sorting and deduplicating assets
+
+`xcm-util` defines a set of functions designed to sort and deduplicate asset collections based on their versioning and properties. The primary goal is to ensure that the assets are organized in a specific order and that any duplicates are removed, while also aggregating certain properties of the assets when duplicates are found.
```

Comment:
I smell ChatGPT again :)

>while also aggregating certain properties of the assets when duplicates are found.

Do you mean the sum of fungibles of the same asset ID? And disregarding duplicate NFTs?

Please rephrase it in a more direct way.